artichoke-core/src/class_registry.rs

```diff
@@ -0,0 +1,40 @@
+use std::any::Any;
+
+pub trait ClassRegistry {
+    type Value;
+    type Error;
+    type Spec;
```

Comment:
What do you think of these docs?

```suggestion
    /// Concrete value type for boxed Ruby values.
    type Value;

    /// Concrete error type for errors encountered when manipulating the class registry.
    type Error;

    /// Type representing a class specification.
    type Spec;
```